bin/flutter

```diff
@@ -185,7 +209,10 @@ fi
 # FLUTTER_TOOL_ARGS="--enable-asserts $FLUTTER_TOOL_ARGS"
 # FLUTTER_TOOL_ARGS="$FLUTTER_TOOL_ARGS --observe=65432"
 
-(upgrade_flutter) 3< "$PROG_NAME"
+# The odd redirect is so that flock has input to the file descriptor it is using
+# for locking. It is ignored if the platform doesn't have flock.
+# The function is defined to run in a subshell.
+upgrade_flutter 7< "$PROG_NAME"
```

Comment:
any reason for `3` -> `7`?

Reply:
Seven is a larger number (and thus less likely to be used by something else). It's really just defensive: 3 is probably fine 99.999% of the time.

Reply:
Besides, seven is lucky. :-)

Reply:
I was 99.999% sure that was the reason...

Reply:
(about it being a larger number)